spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskExecutionServiceTests.java

```diff
@@ -708,6 +709,31 @@ public void validateNullResourceTaskTest() {
 		}
 	}
 
+	@TestPropertySource(properties = { "spring.cloud.dataflow.task.auto-create-task-definitions=true" })
+	@AutoConfigureTestDatabase(replace = Replace.ANY)
+	public static class AutoCreateTaskDefinitionTests extends DefaultTaskExecutionServiceTests {
+
+		@Autowired
+		TaskDefinitionRepository taskDefinitionRepository;
+
+		@Autowired
+		TaskExecutionInfoService taskExecutionInfoService;
+
+		@Before
+		public void setup() {
+			this.launcherRepository.save(new Launcher("default", "local", taskLauncher));
+		}
+
+		@Test
+		@DirtiesContext
+		public void executeTaskWithNullDefinitionCreatesDefinitionIfConfigured() {
+			initializeSuccessfulRegistry(appRegistry);
+			when(this.taskLauncher.launch(any())).thenReturn("0");
+			taskExecutionService.executeTask("demo", new HashMap<>(), new LinkedList<>());
```

Comment:
Can we test the task definitions repo to see if the task definition was created?

Reply:
It couldn't hurt but I don't think we would get to the deployer launch if not.